src/NHSD.BuyingCatalogue.Organisations.Api/Services/MailKitEmailService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using MailKit;
+using NHSD.BuyingCatalogue.Organisations.Api.Settings;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    /// <summary>
+    /// A service for sending e-mails using MailKit.
+    /// </summary>
+    internal sealed class MailKitEmailService : IEmailService
+    {
+        private readonly IMailTransport _client;
+        private readonly SmtpSettings _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MailKitEmailService"/> class
+        /// using the provided <paramref name="client"/> and <paramref name="settings"/>.
+        /// </summary>
+        /// <param name="client">The mail transport to use to send e-mail.</param>
+        /// <param name="settings">The SMTP configuration.</param>
+        public MailKitEmailService(IMailTransport client, SmtpSettings settings)
+        {
+            _client = client;
+
+            // TODO: confirm this is acceptable for us if using SSL
+            // (as we will be using a trusted server that is not
+            // user-configurable this should be fine, providing the
+            // root CA is trusted by the host system)
+            client.ServerCertificateValidationCallback = (s, c, h, e) => true;
+
+            _settings = settings;
```

Comment:
Can't confirm this one unfortunately. I'll ask either @sam-roberts-bjss @CallumBradburyBJSS for a second pair of eyes.

Reply:
The amount of clues I have on this one is zero

Reply:
I guess there are two questions here:

1. Will SSL be required to connect to the SMTP server in any environment?
2. If yes, will the root CA be trusted by the host OS by default?

If the answer is 1 = Yes, 2 = No then we will need to update the callback to handle an `SslHandshakeException` otherwise we don't need to do anything (other than remove the comment).

(We wouldn't need to make any changes to work with TLS. I've tested using a TLS-enabled server and it works as is.)

Maybe one for @stuart-bass-bjss?

Reply:
please push into config. Prod will be SSL, and hopefully trusted, but it's likely local/dev/test will be ssl but self signed. Please use the existing AllowInvalidCertificate config value to set whether invalid ssl certificates are set. Startup already does this for calls to isapi. The recomendation is to set HttpClientHandler.DangerousAcceptAnyServerCertificateValidator as the http handler if you can for this.

Reply:
I could do with testing this. Any suggestions for a quick way to spin up an SMTP server with SSL?

@assoomro do you know if MailDev supports SSL?

Reply:
@bjss-chris not from the top of my head, I'll do some digging and come back to you :]

Reply:
@bjss-chris https://github.com/maildev/maildev#usage there seems to be a flag to enable ssl for outgoing email.
We can raise a task to add this to the config (local docker-compose)

Reply:
@assoomro As that's outgoing I suspect that is for use where you have MailDev set up to relay to messages whereas what we need here is to be able to connect to MailDev using SSL (incoming).

Reply:
You're quite right, Did some more digging and they haven't put that in yet. I'm afraid we won't be able to test this then. You reckon it might be worth to switch over to a SMTP server that provides SSL?

Reply:
To add to my previous response - we can switch to Smtp4Dev which [supports ssl] (https://github.com/smorks/smtp4dev/blob/master/Rnwood.SmtpServer/DefaultServer.cs) Given that we've everything configurable through appsettings.json, it shouldn't take too long to change.

Reply:
I think I'll have a look to see how easy/difficult it is to spin up locally in the meantime so I can start doing some testing.

Reply:
@michaelpace-bjss has confirmed we do need to support TLS 1.2 but not SSL.

The existing implementation already supports TLS and @assoomro has a PR to switch from MailDev to Smtp4dev so this should be fine as is (but I'll remove the TODO comments).

Reply:
Although thinking about it some more I think we do still need to make a tweak to be able to handle cases such as @stuart-bass-bjss mentioned where we will be using self-signed certs – I'll try running some tests...

Reply:
It looks like Smtp4dev defaults to TLS 1.0 so we'll either need to check if there's any configuration we can set or look for an alternative SMTP server for our integration tests.

Reply:
I've updated the implementation so it will allow an untrusted certificate only when the `AllowInvalidCertificate` flag is set in configuration.

There is a flag specifically for the SMTP settings but it will inherit from the app-level settings when not set explicitly.

Reply:
Now all that's need is to make sure test SMTP server supports TLS 1.2 but @assoomro already has that sorted. 👍

Reply:
Update: I've used stunnel as a TLS wrapper for maildev [commit here](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueIdentity/pull/89/commits/aacaf54fe2bc1ebd4fd861fd5b8f93f0e55d3cee)